### PR TITLE
NIFI-12166 Add repo usage to the monitoring endpoint

### PR DESCRIPTION
**Pull request by timeabarna**
# Summary

[NIFI-12166](https://issues.apache.org/jira/browse/NIFI-12166)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-extension-utils/nifi-prometheus-utils/pom.xml [hunk 1]
```diff
@@ -53,6 +53,11 @@
             <artifactId>nifi-metrics</artifactId>
             <version>2.0.0-SNAPSHOT</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-framework-api</artifactId>
+            <version>2.0.0-SNAPSHOT</version>
+        </dependency>
```
Comment (markap14) [inline nifi-nar-bundles/nifi-extension-utils/nifi-prometheus-utils/pom.xml:60]: We should not have extensions / utils depending on nifi-framework-api. That is solely for use by the nifi framework and can change at any point in time.
I would be okay with moving the `StorageUsage` class to nifi-api and adding methods to `EventAccess` to expose the `StorageUsage` for the particular repositories, but we should not be exposing things like SystemDiagnostics to extensions.

Comment (timeabarna) [reply nifi-nar-bundles/nifi-extension-utils/nifi-prometheus-utils/pom.xml]: Thank @markap14 for your review. I've modified the code based on your recommendations.

**Comment by timeabarna** [comment]
Thank @markap14 for your review. I've modified the code based on your recommendations.

**Comment by markap14** [comment]
Thanks @timeabarna looks good to me. +1 will merge.

**Pull request merged**